pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixInstanceDataManager.java

```diff
@@ -180,6 +181,24 @@ private void initInstanceDataDir(File instanceDataDir) {
     }
   }
 
+  @Override
+  public List<File> getConsumerDirPaths() {
+    String consumerDirPath = _instanceDataManagerConfig.getConsumerDir();
+
+    if (consumerDirPath != null) {
+      return Collections.singletonList(new File(consumerDirPath));
+    } else {
+      List<File> consumerDirs = new ArrayList<>();
+      String instanceDataDir = _instanceDataManagerConfig.getInstanceDataDir();
+      for (String tableNameWithType : getAllTables()) {
+        String dirPath = instanceDataDir + File.separator + tableNameWithType + File.separator
```

Comment:
Will this also end up considering offline tables? Do we want to just do this for realtime tables ? If we are doing this for all tables, then it might just be enough to calculate the size of instanceDataDir?

Reply:
It will not count offline table dataDir size as the directory will not exist. Latest commit explicitly checks instance type tho, which should make it clearer.